clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -379,7 +378,12 @@ protected void onJoinComplete(int generation,
 
         // give the assignor a chance to update internal state based on the received assignment
         ConsumerGroupMetadata metadata = new ConsumerGroupMetadata(rebalanceConfig.groupId, generation, memberId, rebalanceConfig.groupInstanceId);
-        assignor.onAssignment(assignment, metadata);
+        assignor.onAssignment(assignment, metadata, revokedPartitions);
```

Comment:
Here we allow the assignor to remove things from `revokedPartitions` to avoid triggering another rebalance if it knows it's not necessary